sopel/tools/__init__.py

```diff
@@ -168,10 +169,31 @@ def decorator(func):
     def deprecated_func(*args, **kwargs):
         if not (warning_in and
                 parse_version(warning_in) >= parse_version(__version__)):
-            stderr(text)
-            # Only display the last frame
+            original_frame = inspect.stack()[-stack_frame]
+            mod = inspect.getmodule(original_frame[0])
+            module_name = None
+            if mod:
+                module_name = mod.__name__
+            if module_name:
+                if 'sopel.' in module_name:
+                    # core, or core plugin
+                    logger = logging.getLogger(module_name)
+                else:
+                    # probably a plugin; use Sopel's public API for getting the
+                    # logger for a plugin
+                    logger = get_logger(module_name.replace('sopel_modules.', ''))
+            else:
+                # don't know the module/plugin name, but we want to make sure
+                # the log line is still output, so fudge it
+                logger = logging.getLogger('sopel.<unknown>')
```

Comment:
Why would you fake it? You can just use the regular logger:

```suggestion
                logger = logging.getLogger(__name__)
```

It would be fine by me.

Reply:
Wouldn't this make the source module appear to be `sopel.tools` or something like that? The point of fudging it is to make the log entry stand out, because this should never be unknown.

Reply:
The traceback + WARNING level should be enough in my opinion.

Reply:
New commit should take care of this lingering concern. 😸